docs/io/fits/index.rst

```diff
@@ -168,11 +168,12 @@ Working with compressed files
 The :func:`open` function will seamlessly open FITS files that have been
 compressed with gzip, bzip2 or pkzip. Note that in this context we are talking
 about a FITS file that has been compressed with one of these utilities (e.g., a
-.fits.gz file).
+.fits.gz file). Opening bzip2-compressed files requires the :mod:`bz2` module
```

Comment:
This makes it sound like it's an optional dependency. I think it's reasonable to expect most users to have the module so this doesn't need to be explicitly mentioned, especially not on the main `index.rst`.

Reply:
Hmm, the whole point of the PR was to make it an optional dependency so users with the (rare) setup of a Python built without `bz2` support would be able to use Astropy with the remaining functionality at all. It is sitting there quite prominently for such a fringe case, but I did not find any other documentation on the automatic (de)compression functionality.

Reply:
>  is sitting there quite prominently for such a fringe case, but I did not find any other documentation on the automatic (de)compression functionality.

Yeah, my suggestion would be to just not add it to the documentation at all.

Reply:
That would probably be the alternative, as most of the `fits` documentation sits in `index.rst`. The exception message is hopefully informative enough whenever this issue might come up in the future.

Reply:
Yes I think we don't need to mention it here, since a normal Python should come with it.